components/common-go/analytics/analytics.go

```diff
@@ -45,7 +45,14 @@ func NewFromEnvironment() Writer {
 		return &logAnalyticsWriter{}
 	case "segment":
 		log.Debug("segment analytics")
-		return &segmentAnalyticsWriter{Client: segment.New(os.Getenv("GITPOD_ANALYTICS_SEGMENT_KEY"))}
+		client, err := segment.NewWithConfig(os.Getenv("GITPOD_ANALYTICS_SEGMENT_KEY"), segment.Config{
+			Endpoint: os.Getenv("GITPOD_ANALYTICS_SEGMENT_ENDPOINT"),
+		})
+		if err != nil {
+			log.WithError(err).Error("cannot create segment client")
```

Comment:
How critical are analytics? Do we want to exit/propagate error if we can't set it up?

Reply:
I cannot answer for everybody. In workspace it is not super critical to interrupt user flow, i.e. if we don't know about usage of gp cli it is not the end of the world. Not sure about other components like dashboard.